Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -1120,7 +1129,41 @@ void UpdatePlanes(std::vector<Frame::Plane::Identifier>& updatedPlanes, std::vec
                 // Clear the list of removed frames to start building up for the next plane update.
                 removedARKitPlanes->clear();
             } @finally {
-                [sessionDelegate UnlockPlanes];
+                [sessionDelegate UnlockAnchors];
+            }
+        }
+        
+        /**
+         Updates existing image anchors in place, gets the list of updated/created image IDs.
+         */
+        void UpdateImageTrackingResults(std::vector<Frame::ImageTrackingResult::Identifier>& updatedImageTrackingResults) {
+            if (![sessionDelegate GetImageDetectionEnabled]) {
+                return;
+            }
+
+            [sessionDelegate LockAnchors];
+            @try {
+                // Iterate over all newly found/updated image anchors and either create or update the image tracking result.
+                auto updatedARImages{[sessionDelegate GetUpdatedImages]};
+                for (ARImageAnchor* updatedImage : *updatedARImages) {
+                    // Update the existing image tracking result if it exists, otherwise create a new result and add it to our list of results.
+                    const auto resultIterator{imageTrackingMap.find({[updatedImage.identifier.UUIDString UTF8String]})};
+                    if (resultIterator != imageTrackingMap.end()) {
+                        UpdateImageTrackingResult(updatedImageTrackingResults, GetImageTrackingResultByID(resultIterator->second), updatedImage);
+                    } else {
+                        // This is a new result. Create it and initialize its values.
+                        imageTrackingResults.push_back(std::make_unique<Frame::ImageTrackingResult>());
+                        auto& result{ *imageTrackingResults.back() };
+                        result.Index = [updatedImage.referenceImage.name intValue];
```

Comment:
This feels slightly sketchy. I guess `name` is not used for anything else and we have full control over it? I kind of would have thought just having an `unordered_map` of image to index would be cleaner, but maybe that would make memory management trickier and more prone to bugs?

Reply:
Name isn't used for anything and we do have full control over it. I went this route because it didn't add any extra members to the class, and since we only ever read it once when first create the image tracking result.

But I can use a map of ARReferenceImage* to index instead.